apiserver/params/params.go

```diff
@@ -481,6 +505,16 @@ type AddApplicationUnitsResults struct {
 
 // AddApplicationUnits holds parameters for the AddUnits call.
 type AddApplicationUnits struct {
+	ApplicationName string                `json:"application"`
```

Comment:
Why not embed the v5 struct
Actually, you don't need a who new struct. Adding the Policy field with omitempty is all that's needed; the new  api  facade will use it, the old one will ignore it. It's no different some client caller sending json  to an api facade with extra fields, we don't complain if there are extras.
In the api client code, you just need reject calls to V5 where Policy != "". I'm pretty sure from memory a similar thing was done with destroy model and storage bool args.

Reply:
We can argue that 'if you're careful you don't need a new struct', but its far easier to make a mistake, or make a mistake with v6 when you don't realize that v5 didn't support it.
I think it is clearer for developers and people introspecting our code base if we have separate types for different versions of our API.